test/lib/recordevents/recorder_vent/recorder.go

```diff
@@ -17,27 +17,101 @@ limitations under the License.
 package recorder_vent
 
 import (
+	"context"
 	"encoding/json"
+	"fmt"
+	"math/rand"
+	"time"
 
 	corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/client-go/tools/record"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	restclient "k8s.io/client-go/rest"
+	kubeclient "knative.dev/pkg/client/injection/kube/client"
+	"knative.dev/pkg/logging"
 
 	"knative.dev/eventing/test/lib/recordevents"
 )
 
 type recorder struct {
-	out record.EventRecorder
-	on  runtime.Object
+	ctx       context.Context
+	namespace string
+	agentName string
+
+	ref *corev1.ObjectReference
 }
 
 func (r *recorder) Vent(observed recordevents.EventInfo) error {
 	b, err := json.Marshal(observed)
 	if err != nil {
 		return err
 	}
+	message := string(b)
+
+	t := time.Now()
+	event := &corev1.Event{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%v.%d", r.ref.Name, observed.Sequence),
+			Namespace: r.namespace,
+		},
+		InvolvedObject: *r.ref,
+		Reason:         recordevents.CloudEventObservedReason,
+		Message:        message,
+		Source:         corev1.EventSource{Component: r.agentName},
+		FirstTimestamp: metav1.Time{Time: t},
+		LastTimestamp:  metav1.Time{Time: t},
```

Comment:
Should we add a comment here about the fact that setting EventTime will ensure in hilarity? Just to make sure somebody doesn't add it inadvertently here?